google_shopping_functions.js

```diff
@@ -1,30 +1,78 @@
-//define getItemsCount function
-//accepts full item data
-//returns the length of the items array
+
 var getItemsCount = function(itemData){
+	for(items in itemsData){
+
+	}
 	return itemData.items.length;
+
 };
+var getItems = function(ObjectData) {
+	return ObjectData.items;
+}
 
-//
-//DEFINE FUNCTIONS HERE
-//
 
+var getItemsByBrand = function (items, brand){
+	var itemsArray= []
+	for(var i=0; i < items.length; i++){
+		if (items[i].product.brand === brand){
+			itemsArray.push(items[i])
+		}
+	}
+	return itemsArray;
+}
 
 
 
+var getItemsCount = function(ObjectData) {
+	return ObjectData.items
+}
 
-//load products.json
-var data = require("./products.json")
 
-//
-//USE FUNCTIONS HERE
-//
+
+
+// var data = require("./products.json")
+
+// var itemsArray=getItems(data);
 
 
 
 
+// console.log('Item Count: ' + getItemsCount(data));
 
+// console.log(getItemsByBrand(itemsArray,"Canon"))
+
+
+var getItemsByAuthor = function (items, author){
+	var itemsArray= []
+	for(var i=0; i < items.length; i++){
+		if (items[i].product.author.name === author){
```

Comment:
For this conditional, we can ignore case using the .toLowerCase() method. Also, we can see if the item's author contains a string of characters using the .indexOf() method, like this:

```
if (items[i].product.author.name.toLowerCase().indexOf(author.toLowerCase()) !== -1){
```

Which will return two Nikon items from an eBay seller in your last console.log.